### PR TITLE
Examples of using is_at_least and providing such handling for TokenV2 collections

### PR DESCRIPTION
**Pull request by igor-aptos**
## Type of Change
- [ ] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [ ] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## Checklist
- [ ] I have read and followed the [CONTRIBUTING](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md) doc
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I identified and added all stakeholders and component owners affected by this change as reviewers
- [ ] I tested both happy and unhappy path of the functionality
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by gelash — APPROVED** [review]

File: aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move [hunk 1]
```diff
@@ -10,6 +10,14 @@
 /// operation that also reduced parallelism, and should be avoided as much as possible.
 /// If you need to capture the value, without revealing it, use snapshot function instead,
 /// which has no parallelism impact.
+///
+/// From paralellism considerations, there are three different levels of effects:
```
Comment (georgemitenkov) [inline aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move:14]: typo: `paralellism` -> `parallelism`

File: aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move [hunk 2]
```diff
@@ -10,6 +10,14 @@
...
+/// * enables full parallelism (cannot create conflicts):
```
Comment (georgemitenkov) [inline aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move:15]: nit: `enable`?

File: aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move [hunk 3]
```diff
@@ -90,31 +98,57 @@ module aptos_framework::aggregator_v2 {
 
     /// Adds `value` to aggregator.
     /// If addition would exceed the max_value, `false` is returned, and aggregator value is left unchanged.
+    ///
+    /// Parallelism info: This operation enables speculative parallelism.
     public native fun try_add<IntElement>(aggregator: &mut Aggregator<IntElement>, value: IntElement): bool;
 
-    // Adds `value` to aggregator, uncoditionally.
-    // If addition would exceed the max_value, EAGGREGATOR_OVERFLOW exception will be thrown.
+    /// Adds `value` to aggregator, uncoditionally.
```
Comment (georgemitenkov) [inline aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move:105]: typo: `uncoditionally` -> `unconditionally`

File: aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move [hunk 4]
```diff
@@ -90,31 +98,57 @@ module aptos_framework::aggregator_v2 {
...
 
     /// Subtracts `value` from aggregator.
     /// If subtraction would result in a negative value, `false` is returned, and aggregator value is left unchanged.
+    ///
+    /// Parallelism info: This operation enables speculative parallelism.
     public native fun try_sub<IntElement>(aggregator: &mut Aggregator<IntElement>, value: IntElement): bool;
 
     // Subtracts `value` to aggregator, uncoditionally.
     // If subtraction would result in a negative value, EAGGREGATOR_UNDERFLOW exception will be thrown.
+    ///
```
Comment (georgemitenkov) [inline aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move:121]: Same here with `uncoditionally` -> `unconditionally`

File: aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move [hunk 5]
```diff
@@ -90,31 +98,57 @@ module aptos_framework::aggregator_v2 {
...
+    /// Parallelism info: This operation enables speculative parallelism.
     public fun sub<IntElement>(aggregator: &mut Aggregator<IntElement>, value: IntElement) {
         assert!(try_sub(aggregator, value), error::out_of_range(EAGGREGATOR_UNDERFLOW));
     }
 
     native fun is_at_least_impl<IntElement>(aggregator: &Aggregator<IntElement>, min_amount: IntElement): bool;
 
+    /// Returns true if aggregator value is larger or equal to the given `min_amount`, false otherwise.
```
Comment (georgemitenkov) [inline aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move:129]: nit: larger than or equal to

File: aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move [hunk 6]
```diff
@@ -90,31 +98,57 @@ module aptos_framework::aggregator_v2 {
...
+    ///
+    /// This operation is more efficient and much more parallelization friendly than calling .
```
Comment (georgemitenkov) [inline aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move:131]: Then calling...? function name is missing or explanation with try_subs

**Review by georgemitenkov — APPROVED** [review]

**Comment by igor-aptos** [comment]
moving changes to DigitalAssets in the Framework for separate and smaller PR, to get relevant people to review, and landing the rest here with current set of reviews

**Pull request merged**